regression/synthetic_dataset.py

```diff
@@ -8,7 +8,7 @@
 import sys
 import urllib.parse
 from argparse import ArgumentParser, Namespace
-from typing import Any, Dict, List, Sequence
```

Comment:
I think for data type annotation, we generally use `List` instead of `list` and `Dict` instead of `dict`. Any reason for changing this?

Reply:
`List` and `Dict` are being deprecated in favor of `list` and `dict`. Python 3.9, PEP 585 https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections